lib/executor/constant_arrival_rate.go

```diff
@@ -227,16 +229,26 @@ func (car ConstantArrivalRate) Run(ctx context.Context, out chan<- stats.SampleC
 	}
 
 	vusFmt := pb.GetFixedLengthIntFormat(maxVUs)
-	fmtStr := pb.GetFixedLengthFloatFormat(arrivalRatePerSec, 2) +
-		" iters/s, " + vusFmt + " out of " + vusFmt + " VUs active"
-
-	progresFn := func() (float64, string) {
+	progresFn := func() (float64, []string) {
 		spent := time.Since(startTime)
 		currentInitialisedVUs := atomic.LoadUint64(&initialisedVUs)
 		vusInBuffer := uint64(len(vus))
-		return math.Min(1, float64(spent)/float64(duration)), fmt.Sprintf(fmtStr,
-			arrivalRatePerSec, currentInitialisedVUs-vusInBuffer, currentInitialisedVUs,
-		)
+		progVUs := fmt.Sprintf(vusFmt+"/"+vusFmt+" VUs",
+			currentInitialisedVUs-vusInBuffer, currentInitialisedVUs)
+		progIters := fmt.Sprintf(
+			pb.GetFixedLengthFloatFormat(arrivalRatePerSec, 0)+" iters/s", arrivalRatePerSec)
+
+		right := []string{progVUs, duration.String(), progIters}
+
+		if spent > duration {
+			return 1, right
+		}
+
+		spentDuration := pb.GetFixedLengthDuration(spent, duration)
+		progDur := fmt.Sprintf("%s/%s", strings.Replace(spentDuration, "s", "", 1), duration)
```

Comment:
Shame for using `strings.Replace()` here, I know, but the alternative would be to reproduce what time.Duration.String() does, which is... a bit more complex. This should be good enough(?). :D

Reply:
`time.Duration` is basically a nonesecond integer, so you can just do something like this:
```go
fmt.Sprintf("%.2f/%s", float64(spentDuration/time.Millisecond)/100, duration)
```

Reply:
ah, sorry, my bad, the `GetFixedLengthDuration` messes this up... then I guess there's not a lot we can do, just forget my initial suggestion - after all, `spendDuration` can also contain minutes and hours